packages/manager/modules/emailpro/src/service/messaging/emailpro.messaging.service.js

```diff
@@ -0,0 +1,169 @@
+    let messageDetails = [];
+    let alertType = 'alert alert-success';
+
+    if (failure != null) {
+      if (failure.message != null) {
+        messageDetails.push({ id: failure.id, message: failure.message });
+        alertType = 'alert alert-warning';
+
+        if (isString(failure.type)) {
+          switch (failure.type.toUpperCase()) {
+            case 'ERROR':
+              alertType += ' alert-danger';
+
+              if (message.ERROR != null || message != null) {
+                messageToSend = message.ERROR || message;
+              }
+              break;
+            case 'PARTIAL':
+              alertType += ' alert-danger';
+
+              if (message.PARTIAL != null || message != null) {
+                messageToSend = message.PARTIAL || message;
+              }
+              break;
+            case 'INFO':
+              alertType += ' alert-success';
+
+              if (message.INFO != null || message != null) {
+                messageToSend = message.INFO || message;
+              }
+              break;
+            default:
+              break;
+          }
+        }
+      } else if (failure.messages != null) {
+        if (!isEmpty(failure.messages)) {
+          alertType = 'alert';
+
+          switch (failure.state.toUpperCase()) {
+            case 'ERROR':
+              alertType += ' alert-danger';
+              messageToSend = message.ERROR || message;
+              break;
+            case 'PARTIAL':
+              alertType += ' alert-warning';
+              messageToSend = message.PARTIAL || message;
+              break;
+            case 'OK':
+              alertType += ' alert-success';
+              messageToSend = message.OK || message;
+              break;
+            default:
+              alertType += ' alert-warning';
+              messageToSend = message;
+              break;
+          }
+
+          messageDetails = failure.messages
+            .filter(
+              (currentMessage) =>
+                isString(currentMessage.type) &&
+                currentMessage.type.toUpperCase() !== 'INFO',
+            )
+            .map((currentMessage) => ({
+              id: currentMessage.code,
+              message: currentMessage.message,
+            }));
+        }
+      } else if (failure.status != null) {
+        alertType = 'alert alert-warning';
+
+        switch (failure.status.toUpperCase()) {
+          case 'BLOCKED':
+          case 'CANCELLED':
+          case 'PAUSED':
+          case 'ERROR':
+            alertType += ' alert-danger';
+            break;
+          case 'TODO':
+          case 'DONE':
+            alertType += ' alert-success';
+            break;
+          default:
+        }
+      } else if (failure === 'true') {
+        alertType = 'alert alert-success';
+        messageDetails = null;
+      }
+    }
+
+    this.message = messageToSend;
+    this.messageDetails = messageDetails;
+    this.alertType = alertType;
```

Comment:
Can be improved with something like that (to avoid `+=` previously)
```suggestion
    this.alertType = `alert ${alertSubType}`;
```

Reply:
Actually I am not quite sure here this is possbile as the classes here are combinations (for instance both warning and danger) cf

![image](https://github.com/ovh/manager/assets/8767496/5932870a-d1d5-4b99-9c25-cd034dfeaf69)